### PR TITLE
Playground thiru

### PR DESCRIPTION
**Pull request by athiruma**

### CODE REVIEW

File: dags/openshift_nightlies/models/release.py [hunk 1]
```diff
@@ -43,8 +43,8 @@ def _generate_cluster_name(self):
 
         if self.platform == 'rosa' or self.platform == 'rogcp' or self.platform == 'hypershift':
             #Only 15 chars are allowed
-            cluster_version = str(self.version).replace(".","")
-            return "perf-"+md5(cluster_name.encode("ascii")).hexdigest()[:3]
+            cluster_version = str(self.version).replace(".","") # removed
+            return "perf-"+md5(cluster_name.encode("ascii")).hexdigest()[:4]
```
Comment (krishvoor) [inline dags/openshift_nightlies/models/release.py]: These changess, why are they needed?

File: dags/openshift_nightlies/manifest.yaml [hunk 2]
```diff
@@ -32,176 +32,12 @@ dagConfig:
...
-      - name: ovnk-data-plane
-        schedule: "0 0 * * 4"
-        config:
-          install: openstack/ovnk.json
-          benchmarks: osp-data-plane.json
-
   # Do not program concurrent builds of ROSA/ROGCP/ARO
   rosa:
-    versions: ["4.12", "4.13"]
+    versions: ["4.13"]
```
Comment (krishvoor) [inline dags/openshift_nightlies/manifest.yaml]: I assume you are doing these to keep things easier to refresh in your playground

File: dags/openshift_nightlies/config/install/rosa/ovn-small.json [hunk 3]
```diff
@@ -8,9 +8,10 @@
     "ocm_environment": "stage",
     "managed_channel_group": "nightly",
     "managed_ocp_version": "latest",
-    "openshift_worker_count": 24,
+    "openshift_worker_count": 1,
```
Comment (krishvoor) [inline dags/openshift_nightlies/config/install/rosa/ovn-small.json]: Can we add another entry `"spot_worker_count": 3`

File: dags/openshift_nightlies/config/install/rosa/ovn-small.json [hunk 4]
```diff
@@ -8,9 +8,10 @@
...
     "openshift_network_type": "OVNKubernetes",
     "openshift_worker_instance_type": "m5.2xlarge",
     "machineset_metadata_label_prefix": "machine.openshift.io",
-    "openshift_workload_node_instance_type": null
+    "openshift_workload_node_instance_type": null,
+    "openshift_spot_workers": true
```
Comment (krishvoor) [inline dags/openshift_nightlies/config/install/rosa/ovn-small.json]: Change this to `"enable_spot_instances": true`

File: dags/openshift_nightlies/scripts/install/rosa.sh [hunk 5]
```diff
@@ -364,6 +364,7 @@ setup(){
     export NETWORK_TYPE=$(cat ${json_file} | jq -r .openshift_network_type)
     export ES_SERVER=$(cat ${json_file} | jq -r .es_server)
     export HCP=$(cat ${json_file} | jq -r .rosa_hcp)
+    export SPOT_WORKERS=$(cat ${json_file} | jq -r .openshift_spot_workers)
```
Comment (krishvoor) [inline dags/openshift_nightlies/scripts/install/rosa.sh]: Update this to reflect to reflect `json` file

File: dags/openshift_nightlies/scripts/install/rosa.sh [hunk 6]
```diff
@@ -476,7 +477,13 @@ install(){
         else
             INSTALLATION_PARAMS="${INSTALLATION_PARAMS} --multi-az"  # Multi AZ is default on hosted-cp cluster
         fi
-        rosa create cluster --tags=User:${GITHUB_USERNAME} --cluster-name ${CLUSTER_NAME} --version "${ROSA_VERSION}" --channel-group=${MANAGED_CHANNEL_GROUP} --compute-machine-type ${COMPUTE_WORKERS_TYPE} --replicas ${COMPUTE_WORKERS_NUMBER} --network-type ${NETWORK_TYPE} ${INSTALLATION_PARAMS} ${ROSA_HCP_PARAMS}
+        if [ "$SPOT_WORKERS" == "true" ]; then
+            rosa create cluster --tags=User:${GITHUB_USERNAME} --cluster-name ${CLUSTER_NAME} --version "${ROSA_VERSION}" --channel-group=${MANAGED_CHANNEL_GROUP} --compute-machine-type ${COMPUTE_WORKERS_TYPE} --network-type ${NETWORK_TYPE} ${INSTALLATION_PARAMS} ${ROSA_HCP_PARAMS}
+            _wait_for_cluster_ready ${CLUSTER_NAME}
+            rosa create machinepool -c ${CLUSTER_NAME} --name="spot-pool" --replicas=${COMPUTE_WORKERS_NUMBER} --instance-type=${COMPUTE_WORKERS_TYPE} –label="rosa-spots=true" --use-spot-instances
```
Comment (krishvoor) [inline dags/openshift_nightlies/scripts/install/rosa.sh:483]: Spot instances will not work for HyperShift, we should tackle it this way.

```
rosa create cluster --tags=User:${GITHUB_USERNAME} --cluster-name ${CLUSTER_NAME} --version "${ROSA_VERSION}" --channel-group=${MANAGED_CHANNEL_GROUP} --compute-machine-type ${COMPUTE_WORKERS_TYPE} --network-type ${NETWORK_TYPE} ${INSTALLATION_PARAMS} ${ROSA_HCP_PARAMS}
        if [ "$SPOT_WORKERS" == "true" ]; then
            _wait_for_cluster_ready ${CLUSTER_NAME}
            rosa create machinepool -c ${CLUSTER_NAME} --name="spot-pool" --replicas=${COMPUTE_WORKERS_NUMBER} --instance-type=${COMPUTE_WORKERS_TYPE} –label="rosa-spots=true" --use-spot-instances
        fi

```

File: dags/openshift_nightlies/tasks/install/rosa/defaults.json [hunk 7]
```diff
@@ -42,3 +42,4 @@
     "aws_region": "us-west-2",
     "oidc_config": ""
 }
+
```
Comment (krishvoor) [inline dags/openshift_nightlies/tasks/install/rosa/defaults.json]: Add the default values:

```
"spot_worker_count": ""
"enable_spot_instances": false
```

**Review by krishvoor — COMMENTED** [review]
Ptal at the comments

File: dags/openshift_nightlies/manifest.yaml (hunk 2 above)
Comment (athiruma) [reply dags/openshift_nightlies/manifest.yaml]: yes

File: dags/openshift_nightlies/scripts/install/rosa.sh (hunk 6 above)
Comment (krishvoor) [reply dags/openshift_nightlies/scripts/install/rosa.sh:483]: As discussed can you ensure the machinepool name as 
`${CLUSTER_NAME}-spot-pool` name

**Review by krishvoor — COMMENTED** [review]
Lemme know, how the DAGs goes

**Pull request closed without merging**